web/src/test/java/edu/softserve/zoo/rest/docs/DocsGenerationTest.java

```diff
@@ -0,0 +1,293 @@
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.mvc.method.RequestMappingInfo;
+import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping;
+
+import java.io.File;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.document;
+import static org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.documentationConfiguration;
+import static org.springframework.restdocs.operation.preprocess.Preprocessors.*;
+import static org.springframework.restdocs.payload.PayloadDocumentation.*;
+import static org.springframework.restdocs.request.RequestDocumentation.parameterWithName;
+import static org.springframework.restdocs.request.RequestDocumentation.pathParameters;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author Taras Zubrei
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration("classpath*:spring/web-test-ctx.xml")
+@WebAppConfiguration
+public class DocsGenerationTest {
+    @Rule
+    public final RestDocumentation restDocumentation = new RestDocumentation(
+            "build/generated-snippets"
+    );
+
+    @Autowired
+    private WebApplicationContext context;
+    @Autowired
+    private RequestMappingHandlerMapping requestMappingHandlerMapping;
+    @Autowired
+    private ParameterNameDiscoverer parameterNameDiscoverer;
+    @Autowired
+    private ObjectMapper objectMapper;
+    private static final Logger LOGGER = LoggerFactory.getLogger(DocsGenerationTest.class);
+
+    private MockMvc mockMvc;
+
+    /**
+     * Set up MockMVC
+     */
+    @Before
+    public void setUp() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.context)
+                .apply(documentationConfiguration(this.restDocumentation))
+                .build();
+    }
+
+    private RestDocumentationResultHandler documentPrettyPrintReqResp(String useCase) {
+        return document(useCase,
+                preprocessRequest(prettyPrint()),
+                preprocessResponse(prettyPrint()));
+    }
+
+    @Test
+    public void run() throws Exception {
+        StringBuilder documentation = new StringBuilder();
+        documentation.append("= Zoo REST API Documentation\n")
```

Comment:
move to properties file or somewhere else. you don't need to append this by stringbuilder